### PR TITLE
APC improvements

### LINKED ISSUE #1294: APCs are scuffed as hell
APC code is just fucked up, there is practically no way to actually repair them if they're broken.

### PR DESCRIPTION
**Pull request by aidan-bowman**
## About the Pull Request

Major code improvements + balance changes + balloon alerts + some other stuff maybe

Fixes #1294 

## Why It's Good For The Game

The code improvements are self explanatory.

Balloon alerts are just plain better for readability. A few users may detest them (i'm looking at you tsuru) but there's a reason most modern codebases use them as much as possible.

Fixing a broken APC is just a little strange. For example, if the APC is broken you're supposed to fix it by... breaking it more. A lot of APC interaction isn't shown to the user at all, making engineering very difficult especially for new players. This isn't a complete fix of them, but it's a step in the right direction.

The duds are moreso because its an underused feature and I think APCs should have more than 4 wires.

## Changelog

:cl:
add: Using a crowbar on an APC with a broken cover now lets you pry off the broken cover.
qol: APCs use balloon alerts instead of chat messages.
qol: Examining APCs gives more useful information.
balance: APCs have 2 dud wires in addition to the 4 functioning wires.
code: APC code improvements.
/:cl:

### CODE REVIEW

File: code/datums/wires/apc.dm [hunk 1]
```diff
@@ -27,25 +27,19 @@
 		if(WIRE_IDSCAN)
 			A.locked = FALSE
 
-			spawn(300)
-				if(A)
-					A.locked = TRUE
+			addtimer(CALLBACK(A, /obj/machinery/power/apc.proc/reset_wire, wire), 30 SECONDS)
```
Comment (Tsurupeta) [inline code/datums/wires/apc.dm:30]: .proc????

File: code/datums/wires/apc.dm [hunk 2]
```diff
@@ -27,25 +27,19 @@
...
 
 		if(WIRE_MAIN_POWER1, WIRE_MAIN_POWER2)
 			if(!A.shorted)
 				A.shorted = TRUE
 
-				spawn(1200)
-					if(A && !is_cut(WIRE_MAIN_POWER1) && !is_cut(WIRE_MAIN_POWER2))
-						A.shorted = FALSE
+				addtimer(CALLBACK(A, /obj/machinery/power/apc.proc/reset_wire, wire), 120 SECONDS)
```
Comment (Tsurupeta) [inline code/datums/wires/apc.dm:36]: .proc?????

File: code/datums/wires/apc.dm [hunk 3]
```diff
@@ -27,25 +27,19 @@
...
 
 		if(WIRE_AI_CONTROL)
 			if(!A.aidisabled)
 				A.aidisabled = TRUE
 
-				spawn(10)
-					if(A && !is_cut(WIRE_AI_CONTROL))
-						A.aidisabled = FALSE
+				addtimer(CALLBACK(A, /obj/machinery/power/apc.proc/reset_wire, wire), 1 SECOND)
```
Comment (Tsurupeta) [inline code/datums/wires/apc.dm:42]: ???

**Review by Tsurupeta — DISMISSED** [review]

File: code/datums/wires/apc.dm (hunk 1 above)
Comment (aidan-bowman) [reply code/datums/wires/apc.dm:30]: What's the issue here?

**Review by EgorDinamit — APPROVED** [review]
Let's merge it!

**Pull request merged**